### PR TITLE
[WIP] Updated applying gradle plugin for flutter_plugin_android_lifecycles

### PR DESCRIPTION
**Pull request by jesswrd**
Updated applying gradle plugins to usage of declarative blocks. Intending on updating all android example apps under packages. Did one as a proof of concept before doing more.

Fixes partially [#152656](https://github.com/flutter/flutter/issues/152656)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/flutter_plugin_android_lifecycle/example/android/settings.gradle [hunk 1]
```diff
@@ -1,28 +1,36 @@
...
+    }
+    dependencies {
+        classpath "gradle.plugin.com.google.cloud.artifactregistry:artifactregistry-gradle-plugin:2.2.1"
     }
-  }
-  dependencies {
-    classpath "gradle.plugin.com.google.cloud.artifactregistry:artifactregistry-gradle-plugin:2.2.1"
-  }
 }
-apply plugin: "com.google.cloud.artifactregistry.gradle-plugin"
```
Comment (reidbaker) [inline packages/flutter_plugin_android_lifecycle/example/android/settings.gradle:28]: I think this apply line is still required

Comment (bartekpacia) [reply packages/flutter_plugin_android_lifecycle/example/android/settings.gradle:28]: Yeah, I agree with Reid. [Source here](https://github.com/flutter/flutter/blob/master/docs/ecosystem/Plugins-and-Packages-repository-structure.md#android-specifics) (I didn't know about it myself).

Fortunately, it seems like that plugin can also be applied in a declarative way :)

https://github.com/GoogleCloudPlatform/artifact-registry-maven-tools?tab=readme-ov-file#gradle-setup

File: packages/flutter_plugin_android_lifecycle/example/android/settings.gradle [hunk 2]
```diff
@@ -1,28 +1,36 @@
...
+
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "8.1.0" apply false
+}
+
+include ":app"
```
Comment (bartekpacia) [inline packages/flutter_plugin_android_lifecycle/example/android/settings.gradle:36]: [I'm pedantic about trailing newlines](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file) :P

**Review by reidbaker — APPROVED** [review]
Please wait to merge after approval from @bartekpacia

**Review by bartekpacia — APPROVED** [review]
overall this looks good, pretty typical migration - apart from the `com.google.cloud.artifactregistry.gradle-plugin`. I'd say that if you don't find a way to make it work with `plugins {}` block within some reasonable time, just keep the old way.

Then a separate issue could be created to track migrating only this specific plugin to declarative. Of course it's all "IMHO" and I defer to @reidbaker :D

**Comment by bartekpacia** [comment]
Hey @jesswrd, I tried applying the artifact-plugin declaratively on my branch in #7811 I think I succeeded. Feel free to copy stuff from there into this PR. I hope this helps.

The only failing test is `Linux repo_checks `, though I believe that that test has to be adjusted because it compares strings/regexes in `build.gradle`.

**Comment by reidbaker** [comment]
@jesswrd before landing this pr Please update the title (at the minimum review WIP)

**Comment by jesswrd** [comment]
> Hey @jesswrd, I tried applying the artifact-plugin declaratively on my branch in #7811 I think I succeeded. Feel free to copy stuff from there into this PR. I hope this helps.
> 
> The only failing test is `Linux repo_checks `, though I believe that that test has to be adjusted because it compares strings/regexes in `build.gradle`.

Oops I merged this PR before I saw these comments. I'm gonna apply those changes on the next plugin then circle back.

**Pull request merged**